lib/pure/math.nim

```diff
@@ -90,7 +90,19 @@ proc isPowerOfTwo*(x: int): bool {.noSideEffect.} =
 proc nextPowerOfTwo*(x: int): int {.noSideEffect.} =
   ## returns `x` rounded up to the nearest power of two.
   ## Zero and negative numbers get rounded up to 1.
-  result = x - 1 
+
+  # Increment x if x is the smallest integer.
+  result = x
+  when defined(cpu64):
```

Comment:
The type is int. I could simply do if(x <0) return 1 but I tried to stick to the approach of this function. It seems faster like that.